ts/features/LandingScreenMultiBanner/store/__tests__/selectors.test.tsx

```diff
@@ -0,0 +1,97 @@
+import { GlobalState } from "../../../../store/reducers/types";
+import { LandingScreenBannerState } from "../reducer";
+import {
+  landingScreenBannerToRenderSelector,
+  localBannerVisibilitySelector
+} from "../selectors";
+
+const mockState = {
+  features: {
+    landingBanners: {
+      ITW_DISCOVERY: true,
+      SETTINGS_DISCOVERY: false
+    }
+  }
+} as GlobalState;
+
+describe("bannerRenderableSelectors", () => {
+  it("should correctly filter between redux visibility and passed selectors, while not breaking if an undefined bannerID is passed by backendStatus, and returning the first valid ID", () => {
```

Comment:
I think we are missing a lot of tests cases for this selector's function. Here my list:
- When `backendBanners` is empty, `localVisibility`has at least one value set to true, with the same entry matching a function that returns true in `renderabilityDataById`, the function should return undefined
- When `backendBanners` has n-entries that do not have any match in either `localVisibility`or `renderabilityDataById`, `localVisibility`has at least one value set to true, with the same entry matching a function that returns true in `renderabilityDataById`, the function should return undefined
- When `backendBanners` has n-entries, with just one and only one that matches another entry in both `localVisibility`AND `renderabilityDataById`, `localVisibility`has at least one value set to true, with the same entry matching a function that returns true in `renderabilityDataById`, the function should return such entry's key

- When `backendBanners` has n-entries with at least one that matches another entry in `renderabilityDataById`, `localVisibility`has no entries, `renderabilityDataById` has at least an entry that returns true and matches the entry in `backendBanners`, the function should return undefined
- When `backendBanners` has n-entries with at least one that matches another entry in `renderabilityDataById`, `localVisibility`has m-entries that do not match with `backendBanners`, `renderabilityDataById` has at least an entry that returns true and matches the entry in `backendBanners`, the function should return undefined
- When `backendBanners` has n-entries with at least one that matches another entry in `renderabilityDataById`, `localVisibility`has m-entries with one that matches `backendBanners` but returns false, `renderabilityDataById` has at least an entry that returns true and matches the entry in `backendBanners`, the function should return undefined
- When `backendBanners` has n-entries with at least one that matches another entry in `renderabilityDataById`, `localVisibility`has m-entries with one that matches `backendBanners` AND returns true, `renderabilityDataById` has at least an entry that returns true and matches the entry in `backendBanners`, the function should return true

- When `backendBanners` has n-entries with at least one that matches another entry in localVisibility, `localVisibility`has at least an entry that returns true and matches the entry in `backendBanners`, `renderabilityDataById` has no entries, the function should return undefined
- When `backendBanners` has n-entries with at least one that matches another entry in localVisibility, `localVisibility`has at least an entry that returns true and matches the entry in `backendBanners`, `renderabilityDataById` has m-entries (all returning true) that do not match with `backendBanners` but at least one matches with one entry in `localVisibility`(for which the returned value is true), the function should return undefined
- When `backendBanners` has n-entries with at least one that matches another entry in localVisibility, `localVisibility`has at least an entry that returns true and matches the entry in `backendBanners`, `renderabilityDataById` has at least an entry (returning true) that matches with one in `backendBanners` but that do not match with localVisibility, the function should return undefined
- When `backendBanners` has n-entries with at least one that matches another entry in localVisibility, `localVisibility`has at least an entry that returns true and matches the entry in `backendBanners`, `renderabilityDataById` has at least an entry (returning true) that matches with one in `backendBanners` but that do not match with localVisibility, the function should return undefined
- When `backendBanners` has n-entries with at least one that matches another entry in localVisibility, `localVisibility`has at least an entry that returns true and matches the entry in `backendBanners`, `renderabilityDataById` has at least an entry (returning true) that matches with one in `backendBanners` and one in `localVisibility`but the latter has a value of false, the function should return undefined
- When `backendBanners` has n-entries with at least one that matches another entry in localVisibility, `localVisibility`has at least an entry that returns true and matches the entry in `backendBanners`, `renderabilityDataById` has at least an entry (returning false) that matches with one in `backendBanners` and one in `localVisibility`with the latter having a value of true, the function should return undefined
- When `backendBanners` has n-entries with at least one that matches another entry in localVisibility, `localVisibility`has at least an entry that returns true and matches the entry in `backendBanners`, `renderabilityDataById` has at least an entry (returning true) that matches with one in `backendBanners` and one in `localVisibility`with the latter having a value of true, the function should return such entry's key